rails/empty-arrays-in-json-request.md

```diff
@@ -0,0 +1,66 @@
+Deep Munging
+
+Even when turned off
+
+```ruby
+put "/api/v1/post/1", {
+  post: { tags: [] },
+  format: :json
+}
+```
+
+
+Will remove the entire `post` hash:
+
+```ruby
+{}
+```
+
+Since the `"Content-Type"` is unset, Rack defaults to treat the request as
+`"application/x-www-form-urlencoded"`.
+
+The URL encoded representation of `["foo", "bar"]` is:
+
+```
+?tags[]=foo&tags[]=bar
+```
+
+However, it isn't possible to represent `[]`:
+
+```
+?tags[]=
+```
+
+As a result, Rack ignores the `tags` key, which would make the `post` key point
+to an empty hash, which Rack also ignores.
```

Comment:
I think what was actually happening was that it wasn't even present in the request body. Since the encoding would do something like:

```ruby
when Array
  values.each do |value|
    encoded_string << "#{key}[]=#{value}"
  end
end
```

which would result in nothing being added to the request body if the array was empty.